### ISSUE TITLE
Mocking magic methods

### ISSUE THREAD

**Issue by dagguh** [hafriedlander/phockito#4]
I can't mock __toString() if the mocked class does not override this magic method.
See http://www.php.net/manual/en/language.oop5.magic.php

**Comment by hafriedlander** [comment]
Ah. Probably not. In fact, you won't be able to mock any method that isn't defined somewhere in the class chain (so you can't create new methods either). 

I'm not sure this is exactly a bug though - it's be nice to be able to do, but mocking a function that isn't defined anywhere feels subtily wrong. I'll have a look how hard it'll be to support though, or at least add a useful error message, principal of least surprise and all.

**Comment by dagguh** [comment]
Tested code:
$logger->log("User $user is rejected");

$user is a dependency implementing a User interface.
RealUser implements User and overrides __toString() so it is logged properly.
In production code I use RealUser (or any other User implementation), but in tests I create a mock of User interface. I cannot stub __toString on it, because it is an interface.

I figured it out, I can put __toString() into User interface and problem should be solved :)
